testing/trino-testing/src/main/java/io/trino/testing/containers/MinioTestContainer.java

```diff
@@ -0,0 +1,107 @@
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.testing.containers;
+
+import com.google.common.collect.ImmutableSet;
+import com.google.common.net.HostAndPort;
+import io.airlift.log.Logger;
+import org.testcontainers.containers.Network;
+
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+
+public class MinioTestContainer
+        extends TestContainer
+{
+    private static final Logger log = Logger.get(MinioTestContainer.class);
+
+    public static final String DEFAULT_IMAGE = "minio/minio:RELEASE.2021-07-15T22-27-34Z";
+    public static final String DEFAULT_HOST_NAME = "minio";
+
+    public static final int MINIO_API_PORT = 4566;
+    public static final int MINIO_CONSOLE_PORT = 4567;
+
+    public static Builder builder()
+    {
+        return new Builder();
+    }
+
+    private MinioTestContainer(
+            String image,
+            String hostName,
+            Set<Integer> exposePorts,
+            Map<String, String> filesToMount,
+            Map<String, String> envVars,
+            Optional<Network> network,
+            int retryLimit)
+    {
+        super(
+                image,
+                hostName,
+                exposePorts,
+                filesToMount,
+                envVars,
+                network,
+                retryLimit);
+    }
+
+    @Override
+    protected void setupContainer()
+    {
+        super.setupContainer();
+        this.container.withCommand(
+                "server",
+                "--address", "0.0.0.0:" + MINIO_API_PORT,
+                "--console-address", "0.0.0.0:" + MINIO_CONSOLE_PORT,
+                "/data");
+    }
+
+    @Override
+    protected void startContainer()
+    {
+        super.startContainer();
+        log.info("MinIO container started with console address: http://" + getMappedMinioConsoleHostAndPort().toString());
+    }
+
+    public HostAndPort getMappedMinioApiHostAndPort()
+    {
+        return getMappedHostAndPortForExposedPort(MINIO_API_PORT);
+    }
+
+    public HostAndPort getMappedMinioConsoleHostAndPort()
+    {
+        return getMappedHostAndPortForExposedPort(MINIO_CONSOLE_PORT);
+    }
+
+    public static class Builder
+            extends TestContainer.Builder<MinioTestContainer.Builder, MinioTestContainer>
+    {
+        private Builder()
+        {
+            this.image = DEFAULT_IMAGE;
+            this.hostName = DEFAULT_HOST_NAME;
+            this.exposePorts =
+                    ImmutableSet.of(
+                            MINIO_API_PORT,
+                            MINIO_CONSOLE_PORT);
```

Comment:
nit: call setters instead accessing fields. Then fields `TestContainer.Builder` can be private.

Reply:
Meh - you still need those for `build` implementation. Does not matter much then I guess.

Reply:
Yes :) i've started to do it and then found my self on adding getters which looks bad in Builder 😄